src/language/parselets.ts

```diff
@@ -109,3 +109,51 @@ export class OperatorParselet implements InfixParselet {
     return { kind: "Binop", op: this.op, op1: left_node, op2: rightNode };
   }
 }
+
+export class BuiltinParselet implements PrefixParselet {
+  parse(tokens: Token[], current_token: Token): Ast {
+    if (current_token.kind != "IDENTIFIER") {
+      throw new Error("Tried to use BuiltinParselet on non-identifier");
+    }
+    const name = current_token.content as Builtin;
+
+    const lparen = tokens.pop();
+    if (!lparen || lparen.kind !== "LPAREN") {
+      throw new Error(`Expected parenthesis after built-in "${name}"`);
+    }
```

Comment:
I think this "expect and consume a certain token" functionality is common enough that we could break it off into a helper function.